server/manager/src/main/java/org/apache/accumulo/manager/upgrade/Upgrader9to10.java

```diff
@@ -726,4 +734,37 @@ static Path resolveRelativeDelete(String oldDelete, String upgradeProperty) {
     }
     return new Path(upgradeProperty, VolumeManager.FileType.TABLE.getDirectory() + oldDelete);
   }
+
+  /**
+   * Remove old temporary map files to prevent problems during recovery.
+   */
+  static void dropSortedMapWALFiles(VolumeManager vm) {
+    Path recoveryDir = new Path("/accumulo/recovery");
```

Comment:
Seems like this should loop over the set of configured volumes instead of this.

Reply:
Do you think using the volumes configured in `instance.volumes` will be enough?

Reply:
Yeah, could look through all of those.  Could narrow it by calling : https://github.com/apache/accumulo/blob/a9100ac35f3d72677b767d7670dfb032c7e7d733/server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManager.java#L180

But I don't think it hurts to just look through all volumes.  Maybe could use https://github.com/apache/accumulo/blob/a9100ac35f3d72677b767d7670dfb032c7e7d733/server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManager.java#L196 to get the list of volumes

Reply:
Would probably be best to avoid choosable() as that will give the volumes configured for new WALs.  The config could change and an old WAL could be on a volume that choosable() no longer returns.  So probably best to inspect all volumes looking for old sorted wals to nuke.

Reply:
> Would probably be best to avoid choosable() as that will give the volumes configured for new WALs. The config could change and an old WAL could be on a volume that choosable() no longer returns. So probably best to inspect all volumes looking for old sorted wals to nuke.

Definitely avoid this. Consider RandomVolumeChooser :smiley_cat:

Reply:
I thought the VolumeChooser was for selecting volumes for writes? I want to loop through every volume configured to find the old sorted WALs to remove. Will the VolumeChooser allow for this?